src/basic/cgroup-util.c

```diff
@@ -1318,6 +1318,28 @@ int cg_pid_get_machine_name(pid_t pid, char **machine) {
         return cg_path_get_machine_name(cgroup, machine);
 }
 
+int cg_path_get_cgroupid(const char *path, uint64_t *ret) {
+        struct file_handle *h;
+        int mnt_id = -1;
+
+        assert(path);
+        assert(ret);
+
+        h = alloca(offsetof(struct file_handle, f_handle) + sizeof(uint64_t));
+
+        h->handle_bytes = sizeof(uint64_t);
+
+        /* This is cgroupfs so we know the size of the handle, thus no need to loop around like
+         * name_to_handle_at_loop() does in mountpoint-util.c */
+        if (name_to_handle_at(AT_FDCWD, path, h, &mnt_id, 0) < 0)
+                return -errno;
+
+        if (ret)
```

Comment:
There's both `assert(ret)` and `if (ret)` - you need only one. If it makes sense to call this without getting the ID back, remove the `assert`, otherwise the `if`.